cobbler/remote.py

```diff
@@ -3842,6 +3844,20 @@ def get_tftp_file(
         self.check_access(token, "get_tftp_file")
         return self.api.get_tftp_file(path, offset, size)
 
+    def __check_items_ready(self, what: str) -> None:
+        """
+        Checking whether the collection items have already been loaded into memory
+        or are not in the process of being loaded.
+
+        :param what: The type of objects.
+        """
+        collection = self.api.get_items(what)
+
+        if not collection.inmemory and collection.deserialize_running:
+            raise CX(
+                f"The {what} collection is not ready yet, please run the command later"
+            )
```

Comment:
This is a comment for documentation: This does not send the status code [`Retry-After`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) with the appropriate error code sadly. This is due to the fact that XML-RPC has no concept for doing so sadly.